packages/fxa-graphql-api/src/backend/legal.service.ts

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { ConfigService } from '@nestjs/config';
+
+import { AppConfig } from '../config';
+import { determineLocale } from 'fxa-shared/l10n/determineLocale';
+import { Injectable } from '@nestjs/common';
+import fetch from 'node-fetch';
+
+const LEGAL_DOCS_PATH = '/settings/legal-docs';
+
+@Injectable()
+export class LegalService {
+  private settingsUrl: string;
+
+  constructor(configService: ConfigService<AppConfig>) {
+    this.settingsUrl = configService.get(
+      'settingsUrl'
+    ) as AppConfig['settingsUrl'];
+  }
+
+  public async getDoc(locale: string, file: string) {
+    if (/^[a-zA-Z-_]{1,500}$/.test(file) === false) {
+      throw new Error(`Invalid file name`);
+    }
+
+    // Determine the best local to use, given the available ones
+    const availableLocales = await this.getAvailableLocales(file);
+    if (!availableLocales) {
+      return { markdown: '' };
+    }
+
+    locale = determineLocale(locale, availableLocales)?.replace('_', '-');
+
+    // Try to get the document from the external legal-docs repo. Note that
+    // since this is an external repo which might be in a transient state some
+    // fallback logic has been added. For example, de-DE would turn into de if
+    // the de-DE document did not exist.
+    let markdown = await this.tryGetDoc(locale, file);
+    if (!markdown && locale !== locale.replace('-.*', '')) {
+      markdown = await this.tryGetDoc(locale.replace('-.*', ''), file);
+    }
+    if (!markdown && locale !== 'en') {
+      markdown = await this.tryGetDoc('en', file);
+    }
+
+    return { markdown };
+  }
+
+  private async tryGetDoc(locale: string, file: string) {
+    const url = `${this.settingsUrl}${LEGAL_DOCS_PATH}/${locale}/${file}.md`;
+    const resp = await fetch(url);
+    const resText = await resp.text();
+
+    // The response should be markdown. Html is returned if the file is not found.
+    if (resText.includes('<!DOCTYPE html>')) {
+      return '';
+    }
+    return resText;
+  }
+
+  private async getAvailableLocales(file: string) {
+    const url = `${this.settingsUrl}${LEGAL_DOCS_PATH}/${file}_locales.json`;
+    const response = await fetch(`${url}`);
+
+    if (!response.ok) {
+      throw Error(response.statusText);
```

Comment:
Nit: should this be _`new Error(...)`_? I know it was copy-pasta from somewhere else (packages/fxa-settings/src/lib/file-utils-legal.tsx#L30) but possibly worth verifying.
```suggestion
      throw new Error(response.statusText);
```

https://eslint.org/docs/latest/rules/no-throw-literal doesn't specifically mention this `throw Error()` syntax, but that makes sense because of the "literal" in the rule name.

Possibly not a big deal since I see we have this pattern 5&times; in the codebase:
```sh
git grep -n "throw Error(" packages | cat

packages/fxa-auth-server/lib/routes/oauth/token.js:197:        throw Error('unreachable');
packages/fxa-event-broker/src/pubsub-proxy/pubsub-proxy.controller.ts:200:        throw Error(`Invalid event: ${message.event}`);
packages/fxa-graphql-api/src/backend/legal.service.ts:67:      throw Error(response.statusText);
packages/fxa-profile-server/lib/events.js:22:      throw Error('Unable get uid from message event ' + message.event);
packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.stories.tsx:50:    throw Error('boop');
```